Jenkinsfile

```diff
@@ -1,29 +1,170 @@
+    }
+
+    testJob = 'no-tests'
+
+    ciTests = {
+      sh("""
+        npm i -g npm@7
+        npm ci
+        npm run test
+      """)
+    }
 
-    buildStep = {
+    buildStep = {cdnUrl ->
         sh("""
-          npm install npm@7 -g && \
-          npm ci && \
-          npm test && \
-          npm run build
+            echo 'CDN_URL ${cdnUrl}'
+            npm --versions
+            npm run build
         """)
     }
 
-    cmConfig = {
-        return [
-            managerEmail: 'purecloud-client-media@genesys.com',
-            rollbackPlan: 'Patch version with fix',
-            qaId: '5d41d9195ca9700dac0ef53a'
-        ]
-    }
+    deployConfig = [
+      dev : 'always',
+      test : 'always',
+      prod : 'always',
+      'fedramp-use2-core': 'always'
+    ]
+
+    onSuccess = {
+      sh("""
+        echo "=== root folder ==="
+        ls -als ./
+
+        echo "=== Printing manifest.json ==="
+        cat ./manifest.json
+
+        echo "=== Printing package.json ==="
+        cat ./package.json
+
+        echo "=== dist folder ==="
+        ls -als dist/
+
+        echo "=== Printing dist/deploy-info.json ==="
+        cat ./dist/deploy-info.json
+
+        # echo "=== Printing dist/package.json ==="
+        # cat ./dist/package.json
 
-    shouldTagOnRelease = { true }
+        echo "=== Printing params ==="
+        echo "ENVIRONMENT  : ${env.ENVIRONMENT}"
+        echo "BUILD_NUMBER : ${env.BUILD_NUMBER}"
+        echo "BUILD_ID     : ${env.BUILD_ID}"
+        echo "BRANCH_NAME  : ${env.BRANCH_NAME}"
+        echo "APP_NAME     : ${env.APP_NAME}"
+        echo "VERSION      : ${env.VERSION}"
+      """)
+
+      def tag = ""
+
+      // if not MAIN branch, then we need to adjust the verion in the package.json
+      if (!isMain()) {
+        // load the package.json version
+        def packageJson = readJSON(file: "./package.json")
+        def featureBranch = env.BRANCH_NAME
+
+        // all feature branches default to --alpha
+        tag = "--tag alpha"
+
+        if (isRelease()) {
+          tag = "--tag next"
+          featureBranch = "release"
+        }
+
+        if (isDevelop()) {
+          tag = "--tag beta"
+          featureBranch = "develop"
+        }
+
+        packageJson.version = "${packageJson.version}-${featureBranch}.${env.BUILD_NUMBER}".toString()
+        println("Writing package.json version to: ${packageJson.version}")
+
+        writeFile(text: new JsonBuilder(packageJson).toPrettyString(), file: './package.json')
+      }
+
+      withCredentials([string(credentialsId: constants.credentials.npm, variable: 'token')]) {
+          sh("""
+              rm -f ~/.npmrc
+              echo "ca=null" > ~/.npmrc
+              echo "//registry.npmjs.org/:_authToken=${token}" >> ~/.npmrc
+
+              cat ./package.json
+              npm publish ${tag} # --dry-run
+          """)
+      }
+
+      // tag, back merge, and prep the next patch verion
+      if (isMain()) {
```

Comment:
This entire function is untested (because we don't want to release anything until it hits master). I don't feel great about that, but we are kind of backed into a corner. 

Most of this code is similar to web-dir prep version release logic.